pkl-core/src/test/kotlin/org/pkl/core/http/HttpClientTest.kt

```diff
@@ -0,0 +1,141 @@
+import org.pkl.commons.test.FileTestUtils
+import org.pkl.core.Release
+import java.net.URI
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.nio.file.Path
+import java.time.Duration
+import kotlin.io.path.copyTo
+import kotlin.io.path.createDirectories
+import kotlin.io.path.createFile
+
+class HttpClientTest {
+  @Test
+  fun `can build default client`() {
+    val client = assertDoesNotThrow {
+      HttpClient.builder().build()
+    } // means we loaded some (built-in) certificates
+    
+    assertThat(client).isInstanceOf(RequestRewritingClient::class.java)
+    client as RequestRewritingClient
+    
+    val release = Release.current()
+    assertThat(client.userAgent).isEqualTo("Pkl/${release.version()} (${release.os()}; ${release.flavor()})")
+    assertThat(client.requestTimeout).isEqualTo(Duration.ofSeconds(60))
+    
+    assertThat(client.delegate).isInstanceOf(JdkHttpClient::class.java)
+    val delegate = client.delegate as JdkHttpClient
+    
+    assertThat(delegate.underlying.connectTimeout()).hasValue(Duration.ofSeconds(60))
+  }
+  
+  @Test
+  fun `can build custom client`() {
+    val client = HttpClient.builder()
+      .setUserAgent("Agent 1")
+      .setRequestTimeout(Duration.ofHours(86))
+      .setConnectTimeout(Duration.ofMinutes(42))  
+      .build() as RequestRewritingClient
+    
+    assertThat(client.userAgent).isEqualTo("Agent 1")
+    assertThat(client.requestTimeout).isEqualTo(Duration.ofHours(86))
+
+    val delegate = client.delegate as JdkHttpClient
+    assertThat(delegate.underlying.connectTimeout()).hasValue(Duration.ofMinutes(42))
+  }
+  
+  @Test
+  fun `can load certificates from file system`() {
+    assertDoesNotThrow {
+      HttpClient.builder().addCertificates(FileTestUtils.selfSignedCertificate).build()
+    }
+  }
+  
+  @Test
+  fun `certificate file located on file system cannot be empty`(@TempDir tempDir: Path) {
+    val file = tempDir.resolve("certs.pem").createFile()
+
+    val e = assertThrows<HttpClientInitException> {
+      HttpClient.builder().addCertificates(file).build()
+    }
+
+    assertThat(e).hasMessageContaining("empty")
+  }
+
+  @Test
+  fun `can load certificates from class path`() {
+    assertDoesNotThrow {
+      HttpClient.builder().addCertificates(javaClass.getResource("IncludedCARoots.pem")).build()
+    }
+  }
+
+  @Test
+  fun `certificate file located on class path cannot be empty`() {
+    val url = javaClass.getResource("emptyCerts.pem")
+
+    val e = assertThrows<HttpClientInitException> {
+      HttpClient.builder().addCertificates(url).build()
+    }
+
+    assertThat(e).hasMessageContaining("empty")
+  }
+
+  @Test
+  fun `can load built-in certificates`() {
+    assertDoesNotThrow {  
+      HttpClient.builder().addBuiltInCertificates().build()
+    }
+  }
+  
+  @Test
+  fun `can load certificates from default location`(@TempDir userHome: Path) {
+    val certFile = userHome.resolve(".pkl")

```

Comment:
I see the intent here, but I feel `userHome` is misleading naming in this test

Reply:
Changed to `tempDir`.